### PR TITLE
[#650] Replace document type with document categories

### PR DESCRIPTION
**Pull request by georgiana-b**
opentrials/opentrials#650

### CODE REVIEW

**Comment by georgiana-b** [comment]
The Travis build is terminated because `make test` stalled, just as in #109, but I have no idea why. Any hypothesis would be appreciated.

File: processors/euctr/extractors.py [hunk 1]
```diff
@@ -132,8 +132,15 @@ def extract_documents(record):
...
-            'type': 'results',
             'source_url': results_url,
         }
         documents.append(document)
     return documents
+
+
+def extract_document_category(record):
+    return {
+        'id': 24,
```
Comment (victor-nitu) [inline processors/euctr/extractors.py:143]: Do we really need IDs for it?

**Review by victor-nitu — COMMENTED** [review]
Looking at this family of related PRs on categories, I was thinking maybe we can upgrade to PostgreSQL 9.5 and use [the new UPSERTs](https://wiki.postgresql.org/wiki/What's_new_in_PostgreSQL_9.5#INSERT_..._ON_CONFLICT_DO_NOTHING.2FUPDATE_.28.22UPSERT.22.29). Pinging @vitorbaptista, does this affect us negatively in any imaginable way?

**Comment by vitorbaptista** [comment]
About the upserts, we already have that functionality thanks to the `dataset` package. Upgrading postgres, as far as I can see, would only make it easier for `dataset` (actually, `sqlalchemy`) to implement this under the hood.

**Comment by victor-nitu** [comment]
@georgiana-b try rebasing master onto this branch and force push, see if it fixes your travis builds.

File: processors/euctr/extractors.py (hunk 1 above)
Comment (georgiana-b) [reply processors/euctr/extractors.py:143]: `document_categories` will have to be kept in sink with Explorer's `data_categories` so we need a common identifier between them. If I choose `name` or [`name`, `group`] as `document_categories` unique identifier, when the `name` will change in `data_categories`, the corresponding `document_category` will not be updated but another one would be be created, leading to discrepancies. So I chose `data_categories` ID.

**Pull request merged**